R/spliceHelperFunctions.R

```diff
@@ -96,232 +96,180 @@ findSpliceOverlapsByDist <-function(query, subject, ignore.strand=FALSE, maxDist
+  tempTable <- tempTable %>% select(alternativeFirstExon, alternativeLastExon, alternativeTSS, alternativeTES)
+  #internal start/end query 
+  queryTable <- annotateInternalStartEnd(queryStartRng, queryEndRng, subjectFullRng, tempTable)
+  colnames(queryTable) <- paste(colnames(queryTable),"query",sep=".")
+  #internal start/end subject
+  subjectTable <- annotateInternalStartEnd(subjectStartRng, subjectEndRng, queryFullRng, tempTable)
+  colnames(subjectTable) <- paste(colnames(subjectTable),"subject",sep=".")
+  tempTable <- cbind(tempTable, queryTable, subjectTable)
+  return (tempTable)
+}
+
+#' annotate internal start and end first exons                                    
+#' @noRd
+annotateInternalStartEnd <- function(StartRng, EndRng, FullRng, alterEndTable){
+  Start.Full.Rng <- expandMatchedRanges(StartRng, FullRng) 
+  End.Full.Rng <- expandMatchedRanges(EndRng, FullRng)
+  #internal start
+  ExonIntersectStart <- pintersect(Start.Full.Rng,mcols(Start.Full.Rng)$matchRng, resolve.empty='start.x')
+  #internal end
+  ExonIntersectEnd=pintersect(End.Full.Rng,mcols(End.Full.Rng)$matchRng, resolve.empty='start.x')
+  tempTable <- tibble(internalFirstExon = tapply(width(ExonIntersectStart), mcols(Start.Full.Rng)$IdMap, sum)!= 0 & alterEndTable$alternativeFirstExon,
+                      internalLastExon = tapply(width(ExonIntersectEnd), mcols(End.Full.Rng)$IdMap, sum)!=0 & alterEndTable$alternativeLastExon)
+  return(tempTable)
+}
+
+#' annotate intron retention 
+#' @noRd
+annotateIntronRetent <- function(Splice.FullSplice.Rng, annotatedTable){
+  IntronRetention <-punion(Splice.FullSplice.Rng, mcols(Splice.FullSplice.Rng)$matchRng, fill.gap=TRUE) == mcols(Splice.FullSplice.Rng)$matchRng
+  outputVector <- tapply(IntronRetention, mcols(Splice.FullSplice.Rng)$IdMap, sum)
+  return (outputVector)
+}
+
+#' annotate exon skiping 
+#' @noRd
+annotateExonSkip <- function(Splice.FullSplice.Rng, Start.Splice.Rng, End.Splice.Rng){
+  ExonSkipping <- punion(Splice.FullSplice.Rng, mcols(Splice.FullSplice.Rng)$matchRng, 
+                         fill.gap=TRUE) == Splice.FullSplice.Rng
+  FirstExonInIntron <- punion(mcols(Start.Splice.Rng)$matchRng, Start.Splice.Rng, 
+                              fill.gap=TRUE) == mcols(Start.Splice.Rng)$matchRng
+  LastExonInIntron <- punion(mcols(Start.Splice.Rng)$matchRng, End.Splice.Rng, 
+                             fill.gap=TRUE) == mcols(Start.Splice.Rng)$matchRng
+  outputVector <- pmax(0, tapply(ExonSkipping, mcols(Splice.FullSplice.Rng)$IdMap, sum) -
+                          tapply(FirstExonInIntron, mcols(Start.Splice.Rng)$IdMap, sum) - 
+                          tapply(LastExonInIntron, mcols(Start.Splice.Rng)$IdMap, sum))
+  return (outputVector)
+}
+
+#' annotate exon splicing
+#' @noRd
+annotateExonSplice <- function(Splice.FullSplice.Rng, Start.Splice.Rng,End.Splice.Rng, strand){
+  ExonStart <- end(Splice.FullSplice.Rng)<end(mcols(Splice.FullSplice.Rng)$matchRng) & 
```

Comment:
this could be made more readable if we follow what is done in the startEnd function and create a dplyr table with the essential information first, and then make these calculations? Or will this be unnecessarily complicated?

Reply:
Two dplyr tables are implemented for calculation in annotateExonSplice().